### PR TITLE
Handle initially empty storage

### PR DESCRIPTION
**Pull request by lemonholic**

### CODE REVIEW

File: consensus/src/lib.rs [hunk 1]
```diff
@@ -160,29 +162,27 @@ impl<N: GossipNetwork, S: Storage> Consensus<N, S> {
...
-                updated_messages: BTreeSet::new(),
-                verified_block_hashes: vec![],
-                vetoed_block_hashes: vec![],
-                finalized: false,
-            };
-            state_storage
-                .add_or_overwrite_file(STATE_FILE_NAME, serde_json::to_string(&state).unwrap())
-                .await?;
-        }
+            commit_state(&mut state_storage, &new_state).await?;
```
Comment (junha1) [inline consensus/src/lib.rs:183]: You have to clear the `state_storage` as well.

File: consensus/src/lib.rs [hunk 2]
```diff
@@ -160,29 +162,27 @@ impl<N: GossipNetwork, S: Storage> Consensus<N, S> {
...
-        let state: State = serde_json::from_str(&state)?;
-        // TODO: check if `this_node_key` is in the validator set. If not, error.
-        if block_header != state.block_header {
-            let height_info = generate_height_info(
-                &block_header,
-                consensus_parameters,
-                round_zero_timestamp,
-                this_node_key.clone(),
-            )?;
+        // Do a pre-calculation for prettier code.
```
Comment (junha1) [inline consensus/src/lib.rs:165]: How about `Prepare new state for the possible storage reset case`

Comment (lemonholic) [reply consensus/src/lib.rs]: applied

File: consensus/src/lib.rs (hunk 1 above)
Comment (lemonholic) [reply consensus/src/lib.rs:185]: fixed

**Review by junha1 — APPROVED** [review]

**Pull request merged**